src/Microsoft.AspNetCore.Razor.Language/Syntax/Syntax.xml

```diff
@@ -27,67 +34,75 @@
+    <Kind Name="MarkupTextLiteral" />
+    <Field Name="LiteralTokens" Type="SyntaxList&lt;SyntaxToken&gt;" />
   </Node>
-  <Node Name="HtmlMarkupBlockSyntax" Base="HtmlSyntaxNode">
-    <Kind Name="HtmlMarkupBlock" />
-    <Field Name="Children" Type="SyntaxList&lt;RazorSyntaxNode&gt;" />
+  <Node Name="MarkupEphemeralTextLiteralSyntax" Base="MarkupSyntaxNode">
+    <Kind Name="MarkupEphemeralTextLiteral" />
+    <Field Name="LiteralTokens" Type="SyntaxList&lt;SyntaxToken&gt;" />
   </Node>
-  <Node Name="HtmlCommentBlockSyntax" Base="HtmlSyntaxNode">
+  <Node Name="HtmlCommentBlockSyntax" Base="RazorBlockSyntax">
     <Kind Name="HtmlCommentBlock" />
-    <Field Name="Children" Type="SyntaxList&lt;RazorSyntaxNode&gt;" />
+    <Field Name="Children" Type="SyntaxList&lt;RazorSyntaxNode&gt;" Override="true" />
   </Node>
-  <Node Name="HtmlTagBlockSyntax" Base="HtmlSyntaxNode">
-    <Kind Name="HtmlTagBlock" />
-    <Field Name="Children" Type="SyntaxList&lt;RazorSyntaxNode&gt;" />
+  <Node Name="MarkupTagBlockSyntax" Base="RazorBlockSyntax">
+    <Kind Name="MarkupTagBlock" />
+    <Field Name="Children" Type="SyntaxList&lt;RazorSyntaxNode&gt;" Override="true" />
   </Node>
-  <Node Name="HtmlMinimizedAttributeBlockSyntax" Base="HtmlSyntaxNode">
+  <Node Name="HtmlMinimizedAttributeBlockSyntax" Base="MarkupSyntaxNode">
     <Kind Name="HtmlMinimizedAttributeBlock" />
-    <Field Name="NamePrefix" Type="HtmlTextLiteralSyntax" Optional="true" />
-    <Field Name="Name" Type="HtmlTextLiteralSyntax" />
+    <Field Name="NamePrefix" Type="MarkupTextLiteralSyntax" Optional="true" />
+    <Field Name="Name" Type="MarkupTextLiteralSyntax" />
   </Node>
-  <Node Name="HtmlAttributeBlockSyntax" Base="HtmlSyntaxNode">
+  <Node Name="HtmlAttributeBlockSyntax" Base="MarkupSyntaxNode">
     <Kind Name="HtmlAttributeBlock" />
-    <Field Name="NamePrefix" Type="HtmlTextLiteralSyntax" Optional="true" />
-    <Field Name="Name" Type="HtmlTextLiteralSyntax" />
-    <Field Name="NameSuffix" Type="HtmlTextLiteralSyntax" Optional="true" />
+    <Field Name="NamePrefix" Type="MarkupTextLiteralSyntax" Optional="true" />
+    <Field Name="Name" Type="MarkupTextLiteralSyntax" />
+    <Field Name="NameSuffix" Type="MarkupTextLiteralSyntax" Optional="true" />
     <Field Name="EqualsToken" Type="SyntaxToken">
       <Kind Name="Equals" />
     </Field>
-    <Field Name="ValuePrefix" Type="HtmlTextLiteralSyntax" Optional="true" />
-    <Field Name="Value" Type="HtmlBlockSyntax" />
-    <Field Name="ValueSuffix" Type="HtmlTextLiteralSyntax" Optional="true" />
+    <Field Name="ValuePrefix" Type="MarkupTextLiteralSyntax" Optional="true" />
+    <Field Name="Value" Type="RazorBlockSyntax" />
+    <Field Name="ValueSuffix" Type="MarkupTextLiteralSyntax" Optional="true" />
   </Node>
-  <Node Name="HtmlLiteralAttributeValueSyntax" Base="HtmlSyntaxNode">
+  <Node Name="HtmlLiteralAttributeValueSyntax" Base="MarkupSyntaxNode">
```

Comment:
Some stuff is 'Markup' and some is still Html?

Reply:
Yes. This is intentional. I named the constructs we know are Html as Html like Attributes, HtmlComments etc

Reply:
so, what's the diff between 'markup' and html?

Reply:
Anything that is parsed in a markup context(read: not CSharp) and not necessarily HTML specific like text literals, Tags(they can also be xml tags) etc. Things that are specific to certain type of markup like HTML, XML etc are named respectively. For example, I can imagine having a node for XMLDeclaration and that will be named `XMLDeclarationSyntax` as opposed to `MarkupDeclarationSyntax`.

Reply:
Right, and attributes are in all forms of markup right?

Reply:
Spoke offline. We'll just name everything Markup. That will always be correct because Markup is a superset of Html. We can revisit and rename certain nodes if there is a need in the future.

Reply:
> Right, and attributes are in all forms of markup right?
> We can revisit and rename certain nodes if there is a need in the future.

Don't see why we made things we know are Html less specific. I disagree with the all-markup naming but don't feel strongly enough to push back. Need a RIP emoji 😉

Reply:
I think whatever you guys talked about here is wicked confusing. Doesn't pass the dan test.